core/timetrace.go

```diff
@@ -234,3 +234,36 @@ func (t *Timetrace) latestNonEmptyDir(dirs []string) (string, error) {
 
 	return "", ErrAllDirectoriesEmpty
 }
+
+// RecordCollides checks if the time of a record collides
+// with other records of the same day and returns a bool
+func (t *Timetrace) RecordCollides(toCheck Record) (bool, error) {
```

Comment:
Glad that we finally have this function. It should be tested though - you can do it yourself if you want to, otherwise I'll write the test.

Reply:
Hm, I am not quite sure how to write the test for it. I would be glad to hear your approach - mocking the fs, or just creating a full core in the test? The easiest idea that came to my mind would be refactoring the part of the code that actually checks for collision to another method and just test this one. In this case, no mocking or fiddling with the core would be needed, right?

Reply:
Yes, you could just move the collision logic in a function that accepts `Record` instances and test that function. The loading of the records will be tested separately.

Reply:
Okay, I just did that and added the tests for it. I think all possible scenarios are tested, but if I missed some it should be easy to add more :)